src/RealtimeEffectPanel.cpp

```diff
@@ -379,6 +402,9 @@ namespace
          if(mProject == nullptr || mEffectState == nullptr)
             return;
 
+         auto& ui = RealtimeEffectStateUI::Get(*mEffectState);
+         ui.Hide();
```

Comment:
When the user deletes an effect from the stack, hide the dialog.

When the user replaces one effect with another in the stack, that too should hide the dialog.  But I don't see a corresponding change in `OnEffectListItemChange`.  That is the function reached through an Observer callback.

Reply:
Will fix

Reply:
Turned out that removing wasn't handled properly either